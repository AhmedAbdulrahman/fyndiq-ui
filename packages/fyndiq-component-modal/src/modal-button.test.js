import React from 'react'
import { shallow } from 'enzyme'
import Button from 'fyndiq-component-button'
import Modal, { ModalButton } from './'

describe('ModalButton Component', () => {
  it('should display a button by default', () => {
    expect(shallow(<ModalButton>Content</ModalButton>)).toMatchSnapshot()
  })

  it('should have a custom button', () => {
    expect(
      shallow(
        <ModalButton button={<Button type="primary">Button</Button>} />,
      ).find(Button),
    ).toMatchSnapshot()
  })

  it('should handle Modal children as a prop', () => {
    expect(
      shallow(
        <ModalButton>
          <Modal overlayClassName="overlayClassName">Content</Modal>
        </ModalButton>,
      ).find(Modal),
    ).toMatchSnapshot()
  })

  it('should open the modal when clicking on the button', () => {
    const Component = shallow(<ModalButton>Content</ModalButton>)
    Component.find(Button).simulate('click')
    Component.update()
    expect(Component.find(Modal).prop('open')).toBe(true)
  })

  it('should close the modal when the modal closes itself', () => {
    const Component = shallow(
      <ModalButton>
        <Modal open>Content</Modal>
      </ModalButton>,
    )
    Component.find(Modal).prop('onClose')()
    Component.update()
    expect(Component.find(Modal).prop('open')).toBe(false)
  })
})
