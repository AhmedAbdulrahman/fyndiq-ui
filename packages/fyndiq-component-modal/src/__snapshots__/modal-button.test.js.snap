// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalButton Component should display a button by default 1`] = `
<span>
  <Button
    className=""
    disabled={false}
    horizontal={false}
    htmlType="button"
    onClick={[Function]}
    pressed={false}
    size="s"
    type="normal"
  >
    Open Modal
  </Button>
  <Modal
    closeClassName=""
    onClose={[Function]}
    open={false}
    overlayClassName=""
    portalId="fyndiq-modal"
    wrapperClassName=""
  >
    Content
  </Modal>
</span>
`;

exports[`ModalButton Component should handle Modal children as a prop 1`] = `
<Modal
  closeClassName=""
  onClose={[Function]}
  open={false}
  overlayClassName="overlayClassName"
  portalId="fyndiq-modal"
  wrapperClassName=""
>
  Content
</Modal>
`;

exports[`ModalButton Component should have a custom button 1`] = `
<Button
  className=""
  disabled={false}
  horizontal={false}
  htmlType="button"
  onClick={[Function]}
  pressed={false}
  size="s"
  type="primary"
>
  Button
</Button>
`;
